crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -201,6 +201,22 @@ def compute_resources_available?
         true
       end
 
+      # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+      def ha_presence_check
```

Comment:
This makes it much more confusing then... Why don't you use the errors hash and return false in this function? At least this is the standard way in rails to do such stuff.